src/precompile_utils.c

```diff
@@ -185,7 +185,7 @@ static int precompile_enq_specialization_(jl_method_instance_t *mi, void *closur
             if (inferred &&
                 inferred != jl_nothing &&
                 jl_ir_flag_inferred((jl_array_t*)inferred) &&
-                (jl_ir_inlining_cost((jl_array_t*)inferred) == UINT16_MAX)) {
+                jl_atomic_load_relaxed(&codeinst->precompile)) {
```

Comment:
Do you think is it okay to use `codeinst->precompile` here?
Now we compute the equivalent of `jl_ir_inlining_cost((jl_array_t*)inferred) == UINT16_MAX)` at Julia-level:
https://github.com/JuliaLang/julia/blob/1120df3e6e83f8b6ff18755b791fea62d6e8bdea/base/compiler/typeinfer.jl#L335-L351